modules/nextflow/src/main/resources/nextflow/executor/command-run.txt

```diff
@@ -99,7 +99,13 @@ nxf_fs_fcp() {
 }
 
 on_exit() {
-    exit_status=${nxf_main_ret:=$?}
+    ## Capture possible errors.
+    ## Can be caused either by the task script, unstage script or after script if defined
+    local last_err=$?
+    ## capture the task error first or fallback to unstage error
+    exit_status=${nxf_main_ret:=0}
+    [[ ${exit_status} -eq 0 && ${nxf_unstage_ret:0} -ne 0 ]] && exit_status=${nxf_unstage_ret}
```

Comment:
Why there's `${nxf_unstage_ret:0}` in the condition and `exit_status=${nxf_unstage_ret}` in the assignment? it may be better to use a local intermediate variable as for `local last_err=$?`

Reply:
This is to decide which exit status to print in the exit_file. First, I check the task ('nxf_main_ret'), then the unstage if exists ('nxf_unstage_ret') and finally the after script ('last_err'). Regarding the local, do you mean make 'exit_status' local?

Reply:
> do you mean make 'exit_status' local?

yes, as long as is not used anywhere else


Let's add extensive comment to describe all this logic that's quite complex